src/transaction/components/TransactionProvider.tsx

```diff
@@ -42,6 +43,14 @@ export function TransactionProvider({
     setTransactionId,
   });
 
+  const { data: callsStatus } = useCallsStatus({
```

Comment:
Let's have the habit that for all Wagmi stuff, do have a custom Hook for it. This can also help us to simplify testing going forward. Because custom Hook are easier to Mock and create test cases.